src/gui/plugins/plotting/Plotting.hh

```diff
@@ -0,0 +1,175 @@
+#include "ignition/gazebo/components/MagneticField.hh"
+#include "ignition/gazebo/components/Model.hh"
+#include "ignition/gazebo/components/Name.hh"
+#include "ignition/gazebo/components/ParentEntity.hh"
+#include "ignition/gazebo/components/ParentLinkName.hh"
+#include "ignition/gazebo/components/Performer.hh"
+#include "ignition/gazebo/components/PerformerAffinity.hh"
+#include "ignition/gazebo/components/Pose.hh"
+#include "ignition/gazebo/components/PoseCmd.hh"
+#include "ignition/gazebo/components/Sensor.hh"
+#include "ignition/gazebo/components/Static.hh"
+#include "ignition/gazebo/components/Visual.hh"
+#include "ignition/gazebo/components/WindMode.hh"
+#include "ignition/gazebo/components/World.hh"
+#include "ignition/gazebo/EntityComponentManager.hh"
+
+namespace ignition {
+
+namespace gazebo {
+
+class PlotComponentPrivate;
+
+class PlotComponent
+{
+  /// \brief Constructor
+  /// \param[in] _type component data type (Pose3d, Vector3d, double)
+  /// \param [in] _entity entity id of that component
+  /// \param [in] _typeId type identifier unique to each component type
+  public: PlotComponent(std::string _type, uint64_t _entity, uint64_t _typeId);
+
+  /// \brief Add a registered chart to the attribute
+  /// \param[in] _attribute component attribute to add the chart to it
+  /// \param[in] _chart chart ID to be added to the attribute
+  public: void RegisterChart(std::string _attribute, int _chart);
+
+  /// \brief Remove a registered chart from the attribute
+  /// \param[in] _attribute component attribute to remove the chart from it
+  /// \param[in] _chart chart ID to be removed from the attribute
+  public: void UnRegisterChart(std::string _attribute, int _chart);
+
+  /// \brief Check if any of the component attributes has any chart
+  /// \return true if any attribute has a chart
+  /// false if all attributes are empty from the charts
+  public: bool HasCharts();
+
+  /// \brief Set a value of specefic component attribute
+  /// \param[in] _attribute component attribute to set its value
+  /// ex : yaw attribute in Pose3d type Component
+  /// \param[in] _value value to be set to the attribute
+  public: void SetAttributeValue(std::string _attribute, const double &_value);
+
+  /// \brief Get all attributes of the component
+  /// \return component attributes
+  public: std::map<std::string, ignition::gui::PlotData*> Data() const;
+
+  /// \brief Get the Component entity ID
+  /// \return entity ID
+  public: uint64_t Entity();
+
+  /// \brief Get the Component type ID
+  /// \return component type ID
+  public: uint64_t TypeId();
+
+  /// \brief dataPtr holds Abstraction data of PlottingPrivate
+  private: std::unique_ptr<PlotComponentPrivate> dataPtr;
+};
+
+class PlottingPrivate;
+
+class Plotting : public ignition::gazebo::GuiSystem
+{
+  Q_OBJECT
+  /// \brief Constructor
+  public: Plotting();
+
+  /// \brief Destructor
+  public: ~Plotting();
+
+    /// \brief called every simulation iteration to access components
```

Comment:
```suggestion
  // Documentation inherited
```

No need to document the inherited functions unless you're adding special functionality to them.